src/control/PageBackgroundChangeController.cpp

```diff
@@ -230,10 +230,10 @@ bool PageBackgroundChangeController::applyPdfBackground(PageRef page)
 		return false;
 	}
 
-	doc->lock();
-	PdfPagesDialog* dlg = new PdfPagesDialog(control->getGladeSearchPath(), doc, control->getSettings());
-	doc->unlock();
-
+	auto* dlg = [this, doc] {
```

Comment:
Why the lambda? Wouldn't a code block suffice?
Actually, it might be better to do a manual unlock ( and lock ?) here ( and similar cases ) for readability instead of relying on scope.

Reply:
Looks like the lambda is used so that `dlg` can be assigned. This looks fine for readability imo.

Reply:
There's no reason not to simply call unlock() before the dlg->show().  
When going out of scope it won't unlock again.
No need for a lambda.

Reply:
Manually locking and unlocking is error-prone (as you can see from the bug) and should be avoided if possible.

Reply:
As @Technius pointed out, the locking / unlocking should only done over an RAII Interface like std::unique_lock, std::lock_guard or std::shared_lock. Without a RAII interface, the locking can break because of any exception or added return without unlocking. 
The lambda is indeed to assign a the variable without leaving it unassigned or setting it to nullptr before. It's like a returning scope block which does not exist in c++.